src/enum_net.cpp

```diff
@@ -764,28 +767,27 @@ int _System __libsocket_sysctl(int* mib, u_int namelen, void *oldp, size_t *oldl
 
 #else
 
-#ifdef _MSC_VER
-#pragma message ( "THIS OS IS NOT RECOGNIZED, enum_net_interfaces WILL PROBABLY NOT WORK" )
-#else
-#warning "THIS OS IS NOT RECOGNIZED, enum_net_interfaces WILL PROBABLY NOT WORK"
-#endif
+#error "Don't know how to enumerate network interfaces on this platform"
 
-		// make a best guess of the interface we're using and its IP
-		udp::resolver r(ios);
-		udp::resolver::iterator i = r.resolve(udp::resolver::query(boost::asio::ip::host_name(ec), "0"), ec);
-		if (ec) return ret;
-		for (;i != udp::resolver::iterator(); ++i)
-		{
-			ip_interface iface;
-			iface.interface_address = i->endpoint().address();
-			if (iface.interface_address.is_v4())
-				iface.netmask = address_v4::netmask(iface.interface_address.to_v4());
-			ret.push_back(iface);
-		}
 #endif
 		return ret;
 	}
 
+	boost::optional<address> get_gateway(ip_interface const& iface, span<ip_route const> routes)
+	{
+		auto const it = std::find_if(routes.begin(), routes.end()
+			, [&](ip_route const& r) -> bool
+			{
+				return r.destination.is_unspecified()
+					&& r.destination.is_v4() == iface.interface_address.is_v4()
+					&& !r.gateway.is_unspecified()
+					&& match_addr_mask(r.gateway, iface.interface_address, iface.netmask)
```

Comment:
This will not work for IPv6. The gateway is typically specified by its link-local address, so this will cause only the link-local address on an interface to have `has_gateway` set which is the exact opposite of what it should be.

Reply:
what does it mean for an address to be "link local" in IPv6? My mental model is based on IPv4, where I would expect all addresses reachable by a switch on the local network would all be considered link local. Just like the IP address associated with the network would be a "local" address (not local as in loopback, but reserved for a private network). The gateway would be on that same private network too. In IPv6, if you have a local network does every node have both a local address and a global address?

I don't have IPv6 connectivity so I don't really have an example of what it looks like. what would be the correct way of ascribing a gateway route to a subnet? where "subnet" means address on the interface + netmask.